Framework/MDAlgorithms/test/ConvertToMDTest.h

```diff
@@ -797,6 +805,47 @@ class ConvertToMDTestPerformance : public CxxTest::TestSuite {
     // this will be used to display progress
     pMockAlgorithm =
         Mantid::Kernel::make_unique<WorkspaceCreationHelper::MockAlgorithm>();
+
+    auto alg = Mantid::API::AlgorithmManager::Instance().createUnmanaged(
+        "CreateSampleWorkspace");
+    alg->initialize();
+    alg->setProperty("WorkspaceType", "Event");
+    alg->setProperty("Function", "Flat background");
+    alg->setProperty("XMin", 10000.0);
+    alg->setProperty("XMax", 100000.0);
+    alg->setProperty("NumEvents", 1000);
+    alg->setProperty("BankPixelWidth", 20);
+    alg->setProperty("Random", false);
+    std::string inWsSampleName = "dummy";
+    alg->setPropertyValue("OutputWorkspace", inWsSampleName);
+    alg->execute();
+
+    std::vector<int> splits(3, 2);
```

Comment:
The construction of these two algorithm instances could probably be simplified. Here are some suggestions. 

* You don't need to go through the `Mantid::API::AlgorithmManager::Instance().createUnmanaged`... you should just be able to create a normal object `ConvertToMD  convertAlgDefault;` Since this has a default constructor, you can also remove the `shared_ptr` needed for the member variables for each algorithm instance.
* The algorithm instances only vary by the "ConverterType" and the "OutputWorkspace". it would be nice if we could setup one instance and just copy-assign it to another, then overwrite those two properties. I assume that is supported, though I have to admit I haven't checked that the base classes support this.
* If your algorithm fails for whatever reason, you might not know about it because  `execute` will swallow exceptions by default. see [here](https://github.com/mantidproject/mantid/blob/master/Framework/API/src/Algorithm.cpp#L672) it might be better to set `alg->setRethrows(true)` to ensure that the original exception propagates out to the test method.

Reply:
Further to my second point above, since you don't need to preserve the "OutputWorkspace" (you never use it) that property can actually be the same between the two algorithms.

Reply:
On a completely different note, it looks like you are clang-formatting a lot to satisfy the build servers. 
you can install `clang-format` locally and hook that into git via [this](https://llvm.org/svn/llvm-project/cfe/trunk/tools/clang-format/git-clang-format) or via your IDE, or just via a simple command ` git diff --name-only | xargs clang-format -i -style="LLVM"`

Reply:
> Further to my second point above, since you don't need to preserve the "OutputWorkspace" (you never use it) that property can actually be the same between the two algorithms.

@OwenArnold If you share the `OutputWorkspace`, the benchmark of the second algorithm execution will probably include the time for deallocating the first one, i.e., the benchmark results get skewed. @igudich I would recommend to *not* share the output.